shared-test/src/main/java/uk/gov/di/authentication/sharedtest/helper/AuditAssertionsHelper.java

```diff
@@ -53,4 +66,32 @@ public static void assertEventTypesReceived(
         // Check that no more events came through while we were looking for the ones we expected.
         assertThat(auditTopic.getCountOfRequests(), equalTo(eventTypes.size()));
     }
+
+    public static void assertTxmaAuditEventsReceived(
+            SqsQueueExtension tmxaAuditQueue, Collection<String> eventTypes) {
+        if (eventTypes.isEmpty()) {
+            throw new RuntimeException(
+                    "Do not call assertTxmaAuditEventsReceived() with an empty collection of event types; it won't wait to see if anything unexpected was received.  Instead, call Thread.sleep and then check the count of requests.");
+        }
+
+        await().atMost(SNS_TIMEOUT, SECONDS)
+                .untilAsserted(
+                        () ->
+                                assertThat(
+                                        tmxaAuditQueue.getApproximateMessageCount(),
+                                        equalTo(eventTypes.size())));
+
+        var receivedEvents =
+                tmxaAuditQueue.getRawMessages().stream()
+                        .peek(System.out::println)
```

Comment:
want to keep that?

Reply:
Good shout, will take this out in next PR.